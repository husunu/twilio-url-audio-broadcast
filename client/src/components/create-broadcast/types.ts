import React from 'react';
import { PropTypes } from '@material-ui/core';

import { StepType, ConfigureStep, RecipientsStep, ReviewStep } from '@/types';
import { ValidatedNumber } from '@/components/create-broadcast/steps/recipients-list/types';

export type StepItem = {
  key: StepType;
  label: string;
  icon: React.ReactNode;
  location: string;
  component: React.FunctionComponent<StepProps>;
  next?: string;
  prev?: string;
  nextButtonColor?: PropTypes.Color;
  nextButtonText?: string;
  nextButtonDisclaimer?: string;
};

export type StepProps = {
  data: ConfigureStep | RecipientsStep | ReviewStep;
  step: StepItem;
};

type NumbersTable = {
  headers: string[];
  rows: Array<keyof ValidatedNumber>;
};

export const VALID_NUMBERS_TABLE: NumbersTable = {
  headers: ['Line', 'Recipients Phone Number'],
  rows: ['line', 'number'],
};

export const INVALID_NUMBERS_TABLE: NumbersTable = {
  headers: ['Line', 'Number', 'Issue'],
  rows: ['line', 'number', 'issue'],
};
