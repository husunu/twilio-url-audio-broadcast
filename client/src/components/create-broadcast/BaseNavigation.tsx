import React from 'react';
import { Box, Grid } from '@material-ui/core';
import { StepItem } from './types';

import {
  BackButton,
  NextButton,
  DiscardBroadcastButton,
} from '@/components/common/buttons';

type BaseNavigationProps = {
  currentStep: StepItem;
  onComplete: () => void;
  canGoNext: boolean;
};

export const BaseNavigation: React.FC<BaseNavigationProps> = ({
  currentStep,
  onComplete = () => {},
  canGoNext,
}) => {
  return (
    <Box mt={2}>
      <Grid container spacing={3}>
        <Grid item xs={8}>
          <DiscardBroadcastButton />
        </Grid>
        <Grid item xs={4} style={{ textAlign: 'right' }}>
          <BackButton prevUrl={currentStep.prev} />
          <NextButton
            nextUrl={currentStep.next}
            onComplete={onComplete}
            disabled={!canGoNext}
          />
        </Grid>
      </Grid>
    </Box>
  );
};
