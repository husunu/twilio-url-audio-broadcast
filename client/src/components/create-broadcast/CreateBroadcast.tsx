import React, { useEffect } from 'react';
import { useCreateBroadcastState } from './hooks';
import { Wizard } from './Wizard';
import { useDispatch } from 'react-redux';
import { discardBroadcast } from '@/redux';

export const CreateBroadcast = () => {
  const { activeStep, data } = useCreateBroadcastState();
  const dispatch = useDispatch();

  // Cleans the Wizard state everytime the Create Notification screen is loaded.
  useEffect(() => {
    dispatch(discardBroadcast());
  }, [dispatch]);

  return <Wizard activeStep={activeStep} data={data} />;
};
