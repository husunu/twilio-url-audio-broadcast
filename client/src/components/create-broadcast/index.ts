export { BaseNavigation } from './BaseNavigation';
export { Wizard } from './Wizard';
export { WizardStep } from './WizardStep';
export { CreateBroadcast } from './CreateBroadcast';
export * from './types';
export { steps } from './constants';
