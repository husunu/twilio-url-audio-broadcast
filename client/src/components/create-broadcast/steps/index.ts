export { Configure } from './configure/Configure';
export { RecipientList } from './recipients-list/RecipientList';
export { Review } from './review/Review';
