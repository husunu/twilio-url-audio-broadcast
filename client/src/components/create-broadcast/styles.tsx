import React from 'react';
import { createStyles, makeStyles } from '@material-ui/core/styles';
import {
  StepConnector,
  StepLabel,
  styled,
  Theme,
  withStyles,
} from '@material-ui/core';

import { Check } from '@material-ui/icons';

interface WizardIconStyleProps {
  active?: boolean;
  completed?: boolean;
  prevStepCompleted?: boolean;
}

interface WizardIconProps extends WizardIconStyleProps {
  icon: React.ReactNode;
}

const useIconStyles = makeStyles((theme) =>
  createStyles({
    step: ({ active, completed }: WizardIconStyleProps) => {
      const mainColor = theme.palette.primary.dark;
      const greyColor = theme.palette.grey[400];
      const color = active || completed ? mainColor : greyColor;
      return {
        color: theme.palette.common.white,
        padding: '10px',
        borderRadius: '30px',
        background: color,
      };
    },
  })
);

export const WizardStepIcon = ({
  icon,
  active,
  completed,
}: WizardIconProps) => {
  const classes = useIconStyles({ active, completed });
  return (
    <div className={classes.step}>
      {completed && !active ? <Check /> : icon}
    </div>
  );
};

export const CustomConnector = withStyles((theme: Theme) => ({
  alternativeLabel: {
    top: 22,
    left: 'calc(-50% + 23px);',
  },
  active: {
    '& $line': {
      borderColor: theme.palette.primary.dark,
    },
  },
  completed: {
    '& $line': {
      borderColor: theme.palette.primary.dark,
    },
  },
  line: {
    borderColor: theme.palette.grey[300],
    borderTopWidth: 3,
    borderRadius: 1,
  },
}))(StepConnector);

export const StyledStepLabel = styled(StepLabel)({
  '& .MuiStepLabel-completed': {
    fontWeight: 'normal',
  },
});
