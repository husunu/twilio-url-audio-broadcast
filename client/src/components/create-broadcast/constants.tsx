import React from 'react';

import { StepItem } from './types';
import { StepType } from '@/types';

import { Configure, RecipientList, Review } from './steps';

import {
  Settings as SettingsIcon,
  GroupAdd as GroupAddIcon,
  Send as SendIcon,
} from '@material-ui/icons';

const locations = {
  configure: '/create/configure',
  recipients: '/create/recipients',
  review: '/create/review',
};

export const steps: Array<StepItem> = [
  {
    key: StepType.Configure,
    label: 'Configure Notification',
    icon: <SettingsIcon />,
    location: locations.configure,
    component: Configure,
    next: locations.recipients,
  },
  {
    key: StepType.Recipients,
    label: 'Upload Recipients',
    icon: <GroupAddIcon />,
    location: locations.recipients,
    component: RecipientList,
    prev: locations.configure,
    next: locations.review,
  },
  {
    key: StepType.Review,
    label: 'Review & Send',
    icon: <SendIcon />,
    location: locations.review,
    component: Review,
    prev: locations.recipients,
    next: '',
  },
];
