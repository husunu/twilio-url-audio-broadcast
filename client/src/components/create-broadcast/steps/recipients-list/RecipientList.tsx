import React, { useMemo } from 'react';
import { useDispatch } from 'react-redux';
import { useRecipientsList } from './hooks';
import { completeRecipientsStep } from '@/redux';
import { RecipientsStep } from '@/types';
import { Box, Typography, Link } from '@material-ui/core';
import { StyledAlert, StyledBox } from './styles';
import { Alert } from '@/components/common/alert';
import { NumbersTable } from '@/components/common/numbers-table';

import useStyles from './styles';
import Uploader from './Uploader';
import Example from './Example';

import {
  Wizard,
  BaseNavigation,
  StepProps,
  VALID_NUMBERS_TABLE,
  INVALID_NUMBERS_TABLE,
} from '@/components/create-broadcast';

export const RecipientList = ({ step, data }: StepProps) => {
  const dispatch = useDispatch();

  const classes = useStyles();

  const {
    numbers: savedNumbers,
    selectedFile: savedSelectedFile,
    completed,
  } = data as RecipientsStep;

  const {
    selectedFile,
    getRootProps,
    getInputProps,
    invalidNumbers,
    validNumbers,
    alert,
  } = useRecipientsList({
    savedNumbers,
    savedSelectedFile,
  });

  const isValid = useMemo(() => {
    return (
      !!selectedFile && invalidNumbers.length < 1 && validNumbers.length > 0
    );
  }, [selectedFile, invalidNumbers, validNumbers]);

  const showResultView = useMemo(() => {
    return completed || selectedFile;
  }, [completed, selectedFile]);

  const onComplete = () => {
    dispatch(
      completeRecipientsStep({
        numbers: validNumbers.map(({ number }: { number: string }) => number),
        selectedFile,
        completed: true,
      })
    );
  };

  return (
    <>
      <Wizard.Step step={step}>
        <Typography variant="h4" component="h2">
          2. Recipients list
        </Typography>
        <Box mt={1}>
          <Typography>
            Please provide the list of recipients to send your voice
            notification.
          </Typography>
        </Box>
        <StyledAlert severity="info">
          Voice notifications are limited to 500 recipients. Please note that
          the destinations you need to call should be enabled. You can manage
          call destinations from Twilio Console in{' '}
          <Link href="https://www.twilio.com/console/voice/calls/geo-permissions">
            Voice Geographic Permission
          </Link>
        </StyledAlert>

        <Box display="flex" flexDirection="row" mt={2}>
          {showResultView && (
            <StyledBox flexGrow={1}>
              <Alert alert={alert} />
              <Box mt={2}>
                {isValid ? (
                  <NumbersTable
                    headers={VALID_NUMBERS_TABLE.headers}
                    rows={VALID_NUMBERS_TABLE.rows}
                    data={validNumbers}
                  />
                ) : (
                  <NumbersTable
                    headers={INVALID_NUMBERS_TABLE.headers}
                    rows={INVALID_NUMBERS_TABLE.rows}
                    data={invalidNumbers}
                  />
                )}
              </Box>
            </StyledBox>
          )}
          <StyledBox
            className={
              !showResultView ? classes.maxUploadView : classes.shrunkUloadVoew
            }
          >
            <Box p={3}>
              <Box display="flex" flexDirection="column" minWidth="400px">
                <Uploader
                  getRootProps={getRootProps}
                  getInputProps={getInputProps}
                  selectedFile={selectedFile}
                />
                <Example />
              </Box>
            </Box>
          </StyledBox>
        </Box>
      </Wizard.Step>
      <BaseNavigation
        currentStep={step}
        onComplete={onComplete}
        canGoNext={isValid}
      />
    </>
  );
};
