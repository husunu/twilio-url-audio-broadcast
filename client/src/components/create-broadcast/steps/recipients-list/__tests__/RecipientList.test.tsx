import React from 'react';
import { Provider } from 'react-redux';
import { HashRouter as Router } from 'react-router-dom';
import { render, act } from '@testing-library/react';
import { RecipientList } from '../RecipientList';
import { store } from '@/redux';
import { StepProps, steps } from '@/components/create-broadcast';

let props: StepProps = {
  step: steps[1],
  data: {
    completed: false,
    numbers: [],
    selectedFile: null,
  },
};

const fullRender = async (componentProps: StepProps) =>
  await render(
    <Provider store={store}>
      <Router>
        <RecipientList {...componentProps} />
      </Router>
    </Provider>
  );

describe('recipients list > RecipientsList', () => {
  test('renders the page', async () => {
    const { asFragment } = await fullRender(props);
    expect(asFragment()).toMatchSnapshot();
  });

  test('renders the padge with saved file', async () => {
    await act(async () => {
      props.data = {
        completed: true,
        numbers: ['+12025555555', '+12025555556', '+12025555557'],
        selectedFile: new File(
          ['+12025555555\n', '+12025555556\n', '+12025555557\n'],
          'test.txt'
        ),
      };
      const { asFragment } = await fullRender(props);
      await new Promise((r) => setTimeout(r, 200));

      expect(asFragment()).toMatchSnapshot();
    });
  });
});
