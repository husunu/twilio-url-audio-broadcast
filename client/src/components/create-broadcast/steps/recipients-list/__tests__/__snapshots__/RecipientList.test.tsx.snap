// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`recipients list > RecipientsList renders the padge with saved file 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiCard-root WithStyles(ForwardRef(Card))-root-295 WithStyles(ForwardRef(Card))-root-296 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiBox-root MuiBox-root-327"
    >
      <h2
        class="MuiTypography-root MuiTypography-h4"
      >
        2. Recipients list
      </h2>
      <div
        class="MuiBox-root MuiBox-root-358"
      >
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          Please provide the list of recipients to send your voice notification.
        </p>
      </div>
      <div
        class="MuiPaper-root MuiAlert-root WithStyles(ForwardRef(Alert))-root-359 MuiAlert-standardInfo MuiPaper-elevation0"
        role="alert"
      >
        <div
          class="MuiAlert-icon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20, 12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10, 10 0 0,0 12,2M11,17H13V11H11V17Z"
            />
          </svg>
        </div>
        <div
          class="MuiAlert-message"
        >
          Voice notifications are limited to 500 recipients. Please note that the destinations you need to call should be enabled. You can manage call destinations from Twilio Console in 
          <a
            class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
            href="https://www.twilio.com/console/voice/calls/geo-permissions"
          >
            Voice Geographic Permission
          </a>
        </div>
      </div>
      <div
        class="MuiBox-root MuiBox-root-391"
      >
        <div
          class="MuiBox-root MuiBox-root-393 Styled(MuiBox)-root-392"
          classes="[object Object]"
        >
          <div
            class="MuiPaper-root MuiAlert-root MuiAlert-standardSuccess MuiPaper-elevation0"
            role="alert"
            style="align-items: center;"
          >
            <div
              class="MuiAlert-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20,12A8,8 0 0,1 12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4C12.76,4 13.5,4.11 14.2, 4.31L15.77,2.74C14.61,2.26 13.34,2 12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0, 0 22,12M7.91,10.08L6.5,11.5L11,16L21,6L19.59,4.58L11,13.17L7.91,10.08Z"
                />
              </svg>
            </div>
            <div
              class="MuiAlert-message"
            >
              The list of recipients is successfully loaded.
            </div>
          </div>
          <div
            class="MuiBox-root MuiBox-root-394"
          >
            <div
              class="MuiBox-root MuiBox-root-397 MuiTableContainer-root makeStyles-tableContainer-395"
            >
              <table
                class="MuiTable-root MuiTable-stickyHeader"
              >
                <thead
                  class="MuiTableHead-root"
                >
                  <tr
                    class="MuiTableRow-root MuiTableRow-head"
                  >
                    <th
                      class="MuiTableCell-root MuiTableCell-head WithStyles(ForwardRef(TableCell))-head-406 MuiTableCell-sizeSmall MuiTableCell-stickyHeader"
                      scope="col"
                    >
                      Line
                    </th>
                    <th
                      class="MuiTableCell-root MuiTableCell-head WithStyles(ForwardRef(TableCell))-head-406 MuiTableCell-sizeSmall MuiTableCell-stickyHeader"
                      scope="col"
                    >
                      Recipients Phone Number
                    </th>
                  </tr>
                </thead>
                <tbody
                  class="MuiTableBody-root"
                >
                  <tr
                    class="MuiTableRow-root"
                  >
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
                    >
                      1
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
                    >
                      +12025555555
                    </td>
                  </tr>
                  <tr
                    class="MuiTableRow-root"
                  >
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
                    >
                      2
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
                    >
                      +12025555556
                    </td>
                  </tr>
                  <tr
                    class="MuiTableRow-root"
                  >
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
                    >
                      3
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
                    >
                      +12025555557
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
            <div
              class="MuiTablePagination-root"
            >
              <div
                class="MuiToolbar-root MuiToolbar-regular MuiTablePagination-toolbar MuiToolbar-gutters"
              >
                <div
                  class="MuiTablePagination-spacer"
                />
                <p
                  class="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
                >
                  1-3 of 3
                </p>
                <div
                  class="MuiTablePagination-actions"
                >
                  <button
                    aria-label="Previous page"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
                    disabled=""
                    tabindex="-1"
                    title="Previous page"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                        />
                      </svg>
                    </span>
                  </button>
                  <button
                    aria-label="Next page"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
                    disabled=""
                    tabindex="-1"
                    title="Next page"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                        />
                      </svg>
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root MuiBox-root-447 Styled(MuiBox)-root-392 makeStyles-shrunkUloadVoew-294"
          classes="[object Object]"
        >
          <div
            class="MuiBox-root MuiBox-root-448"
          >
            <div
              class="MuiBox-root MuiBox-root-449"
            >
              <div
                class="MuiBox-root MuiBox-root-452 makeStyles-uploader-450"
                data-testid="drop-zone"
                tabindex="0"
              >
                <input
                  accept="text/plain"
                  autocomplete="off"
                  style="display: none;"
                  tabindex="-1"
                  type="file"
                />
                <div
                  class="MuiBox-root MuiBox-root-453 makeStyles-uploaderItems-451"
                >
                  <h5
                    class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter"
                  >
                    Drag & Drop Recipient File (.txt)
                  </h5>
                </div>
                <div
                  class="MuiBox-root MuiBox-root-454 makeStyles-uploaderItems-451"
                >
                  <h6
                    class="MuiTypography-root MuiTypography-h6 MuiTypography-alignCenter"
                  >
                    or
                  </h6>
                </div>
                <div
                  class="MuiBox-root MuiBox-root-455 makeStyles-uploaderItems-451"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      Select New File
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </div>
                <div
                  class="MuiBox-root MuiBox-root-485 makeStyles-uploaderItems-451"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1"
                  >
                    test.txt
                  </p>
                </div>
              </div>
              <div
                class="MuiBox-root MuiBox-root-489 makeStyles-root-486"
              >
                <p
                  class="MuiTypography-root makeStyles-title-488 MuiTypography-body1"
                >
                  Example Recipient List
                </p>
                <div
                  class="MuiBox-root MuiBox-root-490"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1"
                  >
                    Text file should contain one phone number per line, and each phone number should be in international format.
                  </p>
                </div>
                <div
                  class="MuiBox-root MuiBox-root-491 makeStyles-numbersBox-487"
                >
                  <ul
                    class="MuiList-root MuiList-dense MuiList-padding"
                  >
                    <li
                      class="MuiListItem-root MuiListItem-dense MuiListItem-gutters"
                    >
                      +15105436767
                    </li>
                    <li
                      class="MuiListItem-root MuiListItem-dense MuiListItem-gutters"
                    >
                      +15107656399
                    </li>
                    <li
                      class="MuiListItem-root MuiListItem-dense MuiListItem-gutters"
                    >
                      +61432286113
                    </li>
                  </ul>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiBox-root MuiBox-root-507"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Discard
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
        style="text-align: right;"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
          style="margin-right: 15px;"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Back
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
          data-testid="next-button"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Next
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`recipients list > RecipientsList renders the page 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiCard-root WithStyles(ForwardRef(Card))-root-3 WithStyles(ForwardRef(Card))-root-4 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiBox-root MuiBox-root-35"
    >
      <h2
        class="MuiTypography-root MuiTypography-h4"
      >
        2. Recipients list
      </h2>
      <div
        class="MuiBox-root MuiBox-root-66"
      >
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          Please provide the list of recipients to send your voice notification.
        </p>
      </div>
      <div
        class="MuiPaper-root MuiAlert-root WithStyles(ForwardRef(Alert))-root-67 MuiAlert-standardInfo MuiPaper-elevation0"
        role="alert"
      >
        <div
          class="MuiAlert-icon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20, 12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10, 10 0 0,0 12,2M11,17H13V11H11V17Z"
            />
          </svg>
        </div>
        <div
          class="MuiAlert-message"
        >
          Voice notifications are limited to 500 recipients. Please note that the destinations you need to call should be enabled. You can manage call destinations from Twilio Console in 
          <a
            class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
            href="https://www.twilio.com/console/voice/calls/geo-permissions"
          >
            Voice Geographic Permission
          </a>
        </div>
      </div>
      <div
        class="MuiBox-root MuiBox-root-99"
      >
        <div
          class="MuiBox-root MuiBox-root-101 Styled(MuiBox)-root-100 makeStyles-maxUploadView-1"
          classes="[object Object]"
        >
          <div
            class="MuiBox-root MuiBox-root-102"
          >
            <div
              class="MuiBox-root MuiBox-root-103"
            >
              <div
                class="MuiBox-root MuiBox-root-106 makeStyles-uploader-104"
                data-testid="drop-zone"
                tabindex="0"
              >
                <input
                  accept="text/plain"
                  autocomplete="off"
                  style="display: none;"
                  tabindex="-1"
                  type="file"
                />
                <div
                  class="MuiBox-root MuiBox-root-107 makeStyles-uploaderItems-105"
                >
                  <h5
                    class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter"
                  >
                    Drag & Drop Recipient File (.txt)
                  </h5>
                </div>
                <div
                  class="MuiBox-root MuiBox-root-108 makeStyles-uploaderItems-105"
                >
                  <h6
                    class="MuiTypography-root MuiTypography-h6 MuiTypography-alignCenter"
                  >
                    or
                  </h6>
                </div>
                <div
                  class="MuiBox-root MuiBox-root-109 makeStyles-uploaderItems-105"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      Select File
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </div>
                <div
                  class="MuiBox-root MuiBox-root-142 makeStyles-uploaderItems-105"
                />
              </div>
              <div
                class="MuiBox-root MuiBox-root-146 makeStyles-root-143"
              >
                <p
                  class="MuiTypography-root makeStyles-title-145 MuiTypography-body1"
                >
                  Example Recipient List
                </p>
                <div
                  class="MuiBox-root MuiBox-root-147"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1"
                  >
                    Text file should contain one phone number per line, and each phone number should be in international format.
                  </p>
                </div>
                <div
                  class="MuiBox-root MuiBox-root-148 makeStyles-numbersBox-144"
                >
                  <ul
                    class="MuiList-root MuiList-dense MuiList-padding"
                  >
                    <li
                      class="MuiListItem-root MuiListItem-dense MuiListItem-gutters"
                    >
                      +15105436767
                    </li>
                    <li
                      class="MuiListItem-root MuiListItem-dense MuiListItem-gutters"
                    >
                      +15107656399
                    </li>
                    <li
                      class="MuiListItem-root MuiListItem-dense MuiListItem-gutters"
                    >
                      +61432286113
                    </li>
                  </ul>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiBox-root MuiBox-root-164"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Discard
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
        style="text-align: right;"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
          style="margin-right: 15px;"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Back
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary Mui-disabled Mui-disabled"
          data-testid="next-button"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Next
          </span>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
